test/Fusion.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE MultiWayIf, TemplateHaskell, TypeApplications #-}
+{-# OPTIONS_GHC -O2 -fplugin Test.Inspection.Plugin #-}
+module Fusion
+( tests
+) where
+
+import Control.Carrier
+import Control.Carrier.Error.Either
+import Control.Carrier.State.Strict
+import Test.Inspection as Inspection
+import Test.Tasty
+import Test.Tasty.HUnit
+
+tests :: TestTree
+tests = testGroup "Effect"
+  [ testCase "eliminates StateCs" $
+    failureOf $(inspectTest $ 'countDown `doesNotUse` ''StateC)
+    @?= Nothing
+  , testCase "eliminates nested StateCs" $
+    failureOf $(inspectTest $ 'countBoth `doesNotUse` ''StateC)
+    @?= Nothing
+  , testCase "eliminates catch and throw" $
+    failureOf $(inspectTest $ 'throwing `doesNotUse` ''ErrorC)
+    @?= Nothing
+  , testCase "eliminates calls to eff" $
+    failureOf $(inspectTest $ 'countDown `doesNotUse` 'eff)
+    @?= Nothing
+  ]
+
+
+failureOf :: Inspection.Result -> Maybe String
+failureOf (Success _) = Nothing
+failureOf (Failure f) = Just f
```

Comment:
I really wanted to be able to use `@?=`, so I made this just return the `String`. We’ll see the error message in failures.